src/components/rating/rating.tsx

```diff
@@ -0,0 +1,177 @@
+   */
+  @Prop() invisibleMode: "collapse" | "keep-space" = "collapse";
+
+  /**
+   * The current value displayed by the component.
+   *
+   */
+  @Prop() maxValue: number;
+
+  /**
+   * This attribute i0ndicates that the user cannot modify the value of the control.
+   * Same as [readonly](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input#attr-readonly)
+   * attribute for `input` elements.
+   * _Disable by default_
+   */
+  @Prop() readonly = false;
+
+  /**
+   * The current value displayed by the component.
+   *
+   */
+  @Prop() value = 0;
+
+  /**
+   * The 'input' event is emitted when a change to the element's value is committed by the user.
+   */
+  @Event() input: EventEmitter;
+
+  onClick(param) {
+    const element = param.target;
+    const score =
+      element.nodeName === "polygon"
+        ? parseInt(element.parentElement.getAttribute("star-number"), 10)
+        : parseInt(element.getAttribute("star-number"), 10);
+    this.ratingScore = score;
+    this.input.emit(this);
+  }
+
+  /**
+   * Returns the id of the inner `input` element (if set).
+   */
+  @Method()
+  async getNativeInputId() {
+    return this.render();
+  }
+
+  private addStarsRating() {
+    const stars = [];
+    for (let i = 0; i < 5; i++) {
+      stars.push(
+        <svg
+          class="rating"
+          viewBox="0 0 100 100"
+          star-number={1 + i}
+          onClick={ev => this.onClick(ev)}
+        >
+          <polygon points="50,0 15,95 100,35 0,35 85,95" />
+        </svg>
+      );
+    }
+    return stars;
+  }
+  private addStarsScore() {
+    const stars = [];
+    let percent: number;
+    let starsScore: number;
+    if (this.maxValue - this.value >= 0) {
+      percent = (this.value * 100) / this.maxValue;
+      starsScore = Math.round((percent * 5) / 100);
+    } else {
+      // tslint:disable-next-line:no-console
+      console.error(
+        "Incongruence in values of 'max-value' and 'value' attributes"
+      );
+    }
+    for (let i = 0; i < 5; i++) {
+      if (i < starsScore) {
+        stars.push(
+          <svg class="score active" viewBox="0 0 100 100" star-number={1 + i}>
+            <polygon points="50,0 15,95 100,35 0,35 85,95" />
+          </svg>
+        );
+      } else {
+        stars.push(
+          <svg class="score" viewBox="0 0 100 100" star-number={1 + i}>
+            <polygon points="50,0 15,95 100,35 0,35 85,95" />
+          </svg>
+        );
+      }
+    }
+    return stars;
+  }
+
+  render() {
+    if (this.readonly) {
+      return this.maxValue && this.value ? (
+        <div>
+          <input
+            type="range"
+            min="0"
+            max={this.maxValue}
+            step="1"
+            value={this.value}
+          />
+          <div class="svgContainer score">{this.addStarsScore()}</div>
+        </div>
+      ) : (
+        // tslint:disable-next-line:no-console
+        console.error(
```

Comment:
Do you want that it reports the exact error? Because I put that error log for show that there is something wrong with the value of `max-value` attribute or `value` attribute

Reply:
Yes, if possible, please.